include/picongpu/plugins/xrayScattering/XrayScatteringWriter.hpp

```diff
@@ -0,0 +1,501 @@
+        ) :
+            fileName( fileName ),
+            dir( dir ),
+            fileExtension( fileExtension ),
+            outputMemoryLayout( outputMemoryLayout),
+            compressionMethod (compressionMethod),
+            globalExtent( globalExtent ),
+            gridSpacing(  gridSpacing  ),
+            unit( unit )
+        {
+            if(outputMemoryLayout ==
+                OutputMemoryLayout::SplitOverNodes)
+            {
+                // Set the MPI communicator.
+                GridController< simDim > & gc =
+                    Environment< simDim >::get().GridController();
+                __getTransactionEvent( ).waitForFinished( );
+                mpiCommunicator = MPI_COMM_NULL;
+                MPI_CHECK( MPI_Comm_dup(
+                    gc.getCommunicator( ).getMPIComm( ),
+                    & mpiCommunicator
+                ) );
+            }
+
+            datatype = openPMD::determineDatatype< T_ValueType >( );
+            // Create the output file.
+            openSeries( openPMD::AccessType::CREATE );
+            openPMDSeries->setMeshesPath( "scatteringData" );
+            openPMDSeries->setAttribute("totalSimulationCells",
+                totalSimulationCells);
+            closeSeries( );
+        }
+
+        virtual ~XrayScatteringWriter( )
+        {
+            if(outputMemoryLayout == OutputMemoryLayout::SplitOverNodes)
+            {
+                if( mpiCommunicator != MPI_COMM_NULL )
+                {
+                    // avoid deadlock between not finished pmacc tasks and mpi
+                    // blocking collectives
+                    __getTransactionEvent( ).waitForFinished( );
+                    MPI_CHECK_NO_EXCEPT(
+                        MPI_Comm_free( &( mpiCommunicator ) ) );
+                }
+            }
+        }
+
+private:
+
+        HINLINE bool isADIOS1()
+        {
+#if openPMD_HAVE_ADIOS1 && !openPMD_HAVE_ADIOS2
+            return this->fileExtension == "bp";
+#else
+            return false;
+#endif
+        }
+
+        /** Opens an openPMD Series in a given access mode.
+         *
+         * @param at OpenPMD API access type.
+         */
+        HINLINE void openSeries( openPMD::AccessType at )
+        {
+            if( !openPMDSeries )
+            {
+                std::string fullName = dir + '/' + fileName +  "." +
+                    fileExtension;
+                log< picLog::INPUT_OUTPUT >(
+                    "XrayScatteringWriter: Opening file: %1%" )
+                % fullName;
+
+                if ( outputMemoryLayout == OutputMemoryLayout::SplitOverNodes )
+                {
+                    // Open a series for a parallel write.
+                    openPMDSeries = std::unique_ptr< openPMD::Series >(
+                        new openPMD::Series(
+                            fullName,
+                            at,
+                            mpiCommunicator
+                        )
+                    );
```

Comment:
Please prefer `pmacc::memory::makeUnique` (like C++14 `std:make_unique`) in such situations as a safer, and also more concise, option.